test/util/test_runners/integration_test_case.py

```diff
@@ -0,0 +1,71 @@
+"""
+extend the unitest TestCase class
+"""
+
+# Builtin
+import io
+import sys
+
+# 3rp parties
+import itertools
+from unittest import TestCase
+from contextlib import redirect_stdout
+import subprocess as sp
+from unittest.mock import patch, MagicMock
+
+# internals
+
+from conjur.cli import Cli
+from test.util.test_runners.params import ClientParams, TestEnvironmentParams
+
+
+class IntegratioTestCaseBase(TestCase):
+
+    def __init__(self, testname, client_params: ClientParams = None,
+                 environmentParams: TestEnvironmentParams = None):
+        """
+        Base class that extends unittest TestCase
+        used to setup tests environment
+        """
+        self.environment = TestEnvironmentParams() if environmentParams is None else environmentParams
+        self.client_params = ClientParams() if client_params is None else client_params
+
+        super(IntegratioTestCaseBase, self).__init__(testname)
+
+
+    def invoke_cli(self, *args, exit_code=0) -> str:
+        """
+        invoker for the integration tests
+        if self.environment.invoke_process it will call the cli binaries to run the integration tests
+        else it will use the Cli code to invoke
+        """
+        if self.environment.invoke_process:
+            return invoke_cli_as_process(self, *args, exit_code=exit_code)
+        return invoke_cli_as_code(self, *args, exit_code=exit_code)
+
+
+def invoke_cli_as_code(test_runner, *args, exit_code=0):
+    capture_stream = io.StringIO()
+    cli_args = list(itertools.chain(*args))
+    with test_runner.assertRaises(SystemExit) as sys_exit:
+        with redirect_stdout(capture_stream):
+            with patch.object(sys, 'argv', ["cli"] + cli_args):
+                Cli().run()
+    print(capture_stream.getvalue())
```

Comment:
don't think you need this print. it might clutter the results

Reply:
removed